lib/ddtrace/contrib/ethon/easy_patch.rb

```diff
@@ -23,14 +27,13 @@ def http_request(url, action_name, options = {})
           end
 
           def headers=(headers)
-            return super unless tracer_enabled?
```

Comment:
I had to remove the `tracer_enabled?` call here and in `reset` because, since we added the resolver pattern, you now have to have a hostname in order to determine the configuration (to decide if the tracer is enabled. theoretically you can now disable the tracer for requests matching `/do-not-want.io/`)... but you can set headers before supplying a URL to net/http.

This will have the side effect of creating a few DD instance variables on the net/http object even if the tracer is disabled.

Reply:
My thought was that it's better to respect all settings per host and leave a few extra ivars on the request object rather than have a surprising situation where you disabled the tracer for certain hosts but they are still getting traced (which would appear to the casual observer to be a bug).

I can certainly derive `enabled?` from the default config if there's no hostname if you feel like the tradeoff isn't worth it.